opm/simulators/wells/MultisegmentWell_impl.hpp

```diff
@@ -262,9 +262,21 @@ namespace Opm
             return;
         }
 
-        BVectorWell xw(1);
-        this->linSys_.recoverSolutionWell(x, xw);
-        updateWellState(simulator, xw, well_state, deferred_logger);
+
+        try{
+            BVectorWell xw(1);
+            this->linSys_.recoverSolutionWell(x, xw);
+
+            updateWellState(simulator, xw, well_state, deferred_logger);
+        }
+        catch(const NumericalProblem& exp) {
+            // Add information about the well and log to deferred logger
+            // (Logging done inside of solve() method will only be seen if
```

Comment:
This confuses me for two reasons:
 - We are not calling solve() here?
 - If we have something in the deferred logger here, it will be seen for any rank, there is a OPM_END_PARALLEL_TRY_CATCH_LOG in BlackoilWellModel<>::recoverWellSolutionAndUpdateWellState().

Please explain, or change comment if wrong.

Reply:
Sorry. The comment is wrong, it was copied from elsewhere without thinking. I will update. Thanks for noticing.

Reply:
Is the comment okay now? As I understand it, we need to catch and rethrow here to handle the throw inside the UMFpack.

Reply:
That is better. Maybe not 100% precise still, as we do not get any logging from UMFPack afaik, just dumped output to standard output? (So it goes to the terminal but not the log files.) But your point is that without the catch and throw here, we might miss an exception thrown due to UMFPack failure?

Reply:
I think so. It is @blattms that added this some time ago for similar calls, so maybe he would like to comment.

Reply:
We can clarify this later, I'll merge as is.